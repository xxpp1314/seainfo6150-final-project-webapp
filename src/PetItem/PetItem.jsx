import React from 'react';
import PropTypes from "prop-types";
import styles from "./PetItem.module.css";
import PetItemImage from '../PetItemImage/PetItemImage';

const PetItem = props => {
    return (
      <div className={styles.petItem}>
        <p  className={styles.text}>{props.petItem.title}</p>
        <PetItemImage url = {props.petItem.img.url} /> 
        <p className={styles.text}>Price: {props.petItem.price}</p>
      </div>
    )
}

PetItem.propTypes = {
    petItem: PropTypes.object.isRequired
};

export default PetItem;