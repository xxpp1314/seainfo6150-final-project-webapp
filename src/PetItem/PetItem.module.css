.petItem{
    font-weight: bolder;
    color: gainsboro;
    align-items: center;
    justify-items: center;
    text-align: center;
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    border-radius: 5%;
  }
  
  
  .petItem:hover {
    border: 5px solid maroon;
    background-color:rgb(99, 54, 54);
    text-decoration-color: maroon;
  }
  
  
  .text {
      padding: 4%;
      color: gainsboro;
      font-weight: bolder;
      text-decoration-line: none;
      text-align: center;
  }