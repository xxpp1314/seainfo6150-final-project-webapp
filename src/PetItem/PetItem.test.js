import React from "react";
import { render } from "@testing-library/react";
import PetItem from "./PetItem.jsx";

const petItem = {
    title: "Russell",
    img:
      {
         "url":"https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/50829956/1/?bust=1619138292&width=1080",
         "height":400,
         "width":500
      },
    price: "$300"
}

it("renders the PetItem correctly", () => {
  const { container } = render(
    <PetItem
      petItem={petItem}
    />
    );
  expect(container).toMatchSnapshot();
});
