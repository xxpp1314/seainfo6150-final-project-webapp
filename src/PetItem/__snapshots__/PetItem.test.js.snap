// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PetItem correctly 1`] = `
<div>
  <div
    class="petItem"
  >
    <p
      class="text"
    >
      Russell
    </p>
    <div>
      <img
        alt=""
        class="image"
        src="https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/50829956/1/?bust=1619138292&width=1080"
      />
    </div>
    <p
      class="text"
    >
      Price: 
      $300
    </p>
  </div>
</div>
`;
